### ISSUE TITLE
THuman Dataset preprocess

### ISSUE THREAD

**Issue by glorioushonor** [YuliangXiu/ICON#189]
Good job!When I run the command of "python -m scripts.render_batch -debug -headless", it gives an error as following

Start Rendering thuman2 with 36 views, 512x512 size.
Output dir: ./debug/thuman2_36views
Rendering types: ['light', 'normal', 'depth']
  0%|                                                                                                                                                                                 | 0/2 [00:00<?, ?it/s]libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
Compilation failure for vertex shader:
b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'
Compilation failure for fragment shader:
b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'
Linker failure: 
b'error: linking with uncompiled/unspecialized shadererror: linking with uncompiled/unspecialized shader'
  0%|                                                                                                                                                                                 | 0/2 [00:03<?, ?it/s]
Traceback (most recent call last):
  File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/media/linjie/2473F3E9F7F20D6D/Code/ICON/scripts/render_batch.py", line 254, in <module>
    for _ in tqdm(
  File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/site-packages/tqdm/std.py", line 1195, in __iter__
    for obj in iterable:
  File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/multiprocessing/pool.py", line 868, in next
    raise value
multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fe4a0bf7a30>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'

How can I fix this?

**Comment by YuliangXiu** [comment]
Have you ever tried to Google the solution?

Such as https://stackoverflow.com/questions/52592309/glsl-3-30-is-not-supported-when-compiling-glsl-shader-on-ubuntu-18-04

**Comment by glorioushonor** [comment]
> 你有没有尝试过谷歌解决方案？
> 
> 如 https://stackoverflow.com/questions/52592309/glsl-3-30-is-not-supported-when-compiling-glsl-shader-on-ubuntu-18-04

Thanks for your reply. 
Yes, I searched for this problem before, and have run "export MESA_GL_VERSION_OVERRIED=3.3" to solve "b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n' ", but there are still errors, which seem to be related to opengl, But I think my computer has installed pyopengl==3.1.1a1 and Pyopengl-Accelerate ==3.1.6 correctly. Could you give me some help, please?


`(ICON) linjie@linjie-Z390-GAMING-X:/media/linjie/2473F3E9F7F20D6D/Code/ICON$ python -m scripts.render_batch -headless -out_dir data/
Start Rendering thuman2 with 36 views, 512x512 size.
Output dir: data//thuman2_36views
Rendering types: ['light', 'normal']
  0%|                                                                                                                                                                               | 0/526 [00:00<?, ?it/s]libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
libEGL warning: DRI2: failed to authenticate
  0%|                                                                                                                                                                               | 0/526 [00:46<?, ?it/s]
Traceback (most recent call last):
  File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/media/linjie/2473F3E9F7F20D6D/Code/ICON/scripts/render_batch.py", line 254, in <module>
    for _ in tqdm(
  File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/site-packages/tqdm/std.py", line 1195, in __iter__
    for obj in iterable:
  File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/multiprocessing/pool.py", line 868, in next
    raise value
multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fb463cc16a0>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'`

**Comment by glorioushonor** [comment]
> 你有没有尝试过谷歌解决方案？
> 
> 如 https://stackoverflow.com/questions/52592309/glsl-3-30-is-not-supported-when-compiling-glsl-shader-on-ubuntu-18-04

I'm sorry to bother you again. Aiming at solving these problems, I've been browsing all of Google and issues in your GitHub, but I still can't fix the problem which seem to be related to opengl and multiprocessing. I can't seem to go on, and it seems that a lot of people have problems here. Could you provide the synthetic dataset for training, or give some suggesions please.

**Comment by YuliangXiu** [comment]
You could use the old rendering code, see the old [dataset.md](https://github.com/YuliangXiu/ICON/blob/74b5c7db64b04a13d5b921504921f2f4231119c5/docs/dataset.md)

**Comment by glorioushonor** [comment]
> You could use the old rendering code, see the old [dataset.md](https://github.com/YuliangXiu/ICON/blob/74b5c7db64b04a13d5b921504921f2f4231119c5/docs/dataset.md) and old [scripits](https://github.com/YuliangXiu/ICON/tree/74b5c7db64b04a13d5b921504921f2f4231119c5/scripts)

Thank you very much! The older version of the script worked fine, and I was able to fix some minor problems. But to run the old script correctly, I change the ms_rate=16 to ms_rate=1.0 in line 115 of render_single.py. I already know that this setting is bad for anti-aliasing, but my computer can only achieve `ms_rate=1.0`, otherwise the program will report an error. Should I switch to a higher performance GPU? Is the second script running long? I've been running it for half a day (1080 Ti).
BTW, congratulations on your paper ECON being accepted by CVPR2023.

**Comment by glorioushonor** [comment]
> You could use the old rendering code, see the old [dataset.md](https://github.com/YuliangXiu/ICON/blob/74b5c7db64b04a13d5b921504921f2f4231119c5/docs/dataset.md) and old [scripits](https://github.com/YuliangXiu/ICON/tree/74b5c7db64b04a13d5b921504921f2f4231119c5/scripts)

I'm sorry to bother you again. 
I found the rendering process time-consuming. After browsing the script file, I find that two Gpus are used to run the script. I wonder if my understanding is correct.

**Comment by YuliangXiu** [comment]
One GPU is enough, if it's too slow, maybe https://github.com/YuliangXiu/ICON/issues/62#issuecomment-1134315213 is related to your issue.

**Comment by glorioushonor** [comment]
> 



> And to run the old script correctly, I change the ms_rate=16 to ms_rate=1.0 in line 115 of render_single.py. I already know that this setting is bad for anti-aliasing, but my computer can only achieve `ms_rate=1.0`, otherwise the program will report an error. Should I switch to a higher performance GPU?

Thanks to your kind reply.
I think it's working correctly. I used a 1080Ti GPU and the script has been running for 24 hours and has rendered about half of the data set. How long does the rendering process take on the Quadro RTX 5000 which you used?
In addition,  to run the old script correctly,  I change the ms_rate=16 to ms_rate=1.0 in line 115 of render_single.py. I already know that this setting is bad for anti-aliasing, but my computer can only achieve ms_rate=1.0, otherwise the program will report an error.
To sum up, I'm going to use a single 3090 for the render and rerender process. Am I right?

**Comment by YuliangXiu** [comment]
It's kind of slow, 36 views per scan only takes me a few hours for rendering. You can use `cprofile` to check the performance bottleneck.

I guess the `ms_rate=1.0` won't affect your final results too much.

**Comment by glorioushonor** [comment]
> You could use the old rendering code, see the old [dataset.md](https://github.com/YuliangXiu/ICON/blob/74b5c7db64b04a13d5b921504921f2f4231119c5/docs/dataset.md) and old [scripits](https://github.com/YuliangXiu/ICON/tree/74b5c7db64b04a13d5b921504921f2f4231119c5/scripts)

Sorry to bother you again, and thank you for your help before. I previously rendered the THuman2.0 dataset by running the old version of the script, but the preprocessed data cannot aligned to the smplx mesh. The results are as follows:
![屏幕截图 2023-03-26 180749](https://user-images.githubusercontent.com/120625228/227939903-73ce40b5-ff2c-4d12-82fd-fd2d87bbffe5.jpg)
I noticed by browsing GitHub that you suggested a new script to solve this problem. But now I can't run through the new script, is there any way to correct the old script to make the result correct instead of using the new script? Could you please give me some advice, thank you.

**Comment by MalignusCN** [comment]
> Good job! When I run the command of "python -m scripts.render_batch -debug -headless", it gives an error as following:
> 
> ```shell
> Star /t Rendering thuman2 with 36 views, 512x512 size.
> Output dir: ./debug/thuman2_36views
> Rendering types: ['light', 'normal', 'depth']
>   0%|                                                                                                                                                                                 | 0/2 [00:00<?, ?it/s]libEGL warning: DRI2: failed to authenticate
> libEGL warning: DRI2: failed to authenticate
> Compilation failure for vertex shader:
> b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'
> Compilation failure for fragment shader:
> b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'
> Linker failure: 
> b'error: linking with uncompiled/unspecialized shadererror: linking with uncompiled/unspecialized shader'
>   0%|                                                                                                                                                                                 | 0/2 [00:03<?, ?it/s]
> Traceback (most recent call last):
>   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 194, in _run_module_as_main
>     return _run_code(code, main_globals, None,
>   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 87, in _run_code
>     exec(code, run_globals)
>   File "/media/linjie/2473F3E9F7F20D6D/Code/ICON/scripts/render_batch.py", line 254, in <module>
>     for _ in tqdm(
>   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/site-packages/tqdm/std.py", line 1195, in __iter__
>     for obj in iterable:
>   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/multiprocessing/pool.py", line 868, in next
>     raise value
> multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fe4a0bf7a30>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'
> ```
> 
> How can I fix this?

I've found the same issue here, directly use 
`export MESA_GL_VERSION_OVERRIDE=3.3`
doesn't useful for me, but with os.environ is useful

**Comment by glorioushonor** [comment]
> > Good job! When I run the command of "python -m scripts.render_batch -debug -headless", it gives an error as following:
> > ```shell
> > Star /t Rendering thuman2 with 36 views, 512x512 size.
> > Output dir: ./debug/thuman2_36views
> > Rendering types: ['light', 'normal', 'depth']
> >   0%|                                                                                                                                                                                 | 0/2 [00:00<?, ?it/s]libEGL warning: DRI2: failed to authenticate
> > libEGL warning: DRI2: failed to authenticate
> > Compilation failure for vertex shader:
> > b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'
> > Compilation failure for fragment shader:
> > b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'
> > Linker failure: 
> > b'error: linking with uncompiled/unspecialized shadererror: linking with uncompiled/unspecialized shader'
> >   0%|                                                                                                                                                                                 | 0/2 [00:03<?, ?it/s]
> > Traceback (most recent call last):
> >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 194, in _run_module_as_main
> >     return _run_code(code, main_globals, None,
> >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 87, in _run_code
> >     exec(code, run_globals)
> >   File "/media/linjie/2473F3E9F7F20D6D/Code/ICON/scripts/render_batch.py", line 254, in <module>
> >     for _ in tqdm(
> >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/site-packages/tqdm/std.py", line 1195, in __iter__
> >     for obj in iterable:
> >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/multiprocessing/pool.py", line 868, in next
> >     raise value
> > multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fe4a0bf7a30>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > How can I fix this?
> 
> I've found the same issue here, directly use `export MESA_GL_VERSION_OVERRIDE=3.3` doesn't useful for me, but with os.environ is useful

Hello, thanks for your kind sharing. Whether you use `os.environ` to solve
`b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'`
or another error
`multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fe4a0bf7a30>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'`
I am still looking for a solution to the second error. Could you please give me some help.

**Comment by yangqinhui0423** [comment]
> > > Good job! When I run the command of "python -m scripts.render_batch -debug -headless", it gives an error as following:
> > > ```shell
> > > Star /t Rendering thuman2 with 36 views, 512x512 size.
> > > Output dir: ./debug/thuman2_36views
> > > Rendering types: ['light', 'normal', 'depth']
> > >   0%|                                                                                                                                                                                 | 0/2 [00:00<?, ?it/s]libEGL warning: DRI2: failed to authenticate
> > > libEGL warning: DRI2: failed to authenticate
> > > Compilation failure for vertex shader:
> > > b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'
> > > Compilation failure for fragment shader:
> > > b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'
> > > Linker failure: 
> > > b'error: linking with uncompiled/unspecialized shadererror: linking with uncompiled/unspecialized shader'
> > >   0%|                                                                                                                                                                                 | 0/2 [00:03<?, ?it/s]
> > > Traceback (most recent call last):
> > >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 194, in _run_module_as_main
> > >     return _run_code(code, main_globals, None,
> > >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 87, in _run_code
> > >     exec(code, run_globals)
> > >   File "/media/linjie/2473F3E9F7F20D6D/Code/ICON/scripts/render_batch.py", line 254, in <module>
> > >     for _ in tqdm(
> > >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/site-packages/tqdm/std.py", line 1195, in __iter__
> > >     for obj in iterable:
> > >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/multiprocessing/pool.py", line 868, in next
> > >     raise value
> > > multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fe4a0bf7a30>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > How can I fix this?
> > 
> > 
> > I've found the same issue here, directly use `export MESA_GL_VERSION_OVERRIDE=3.3` doesn't useful for me, but with os.environ is useful
> 
> Hello, thanks for your kind sharing. Whether you use `os.environ` to solve `b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'` or another error `multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fe4a0bf7a30>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'` I am still looking for a solution to the second error. Could you please give me some help.

Hey, I want to know if you have solved the second error. I also met the same error and have tried all the solutions above but it's not solved.
My error is as follows:
_(icon) ciciyang@DESKTOP-RVT20TR:~/code/ICON$ python -m scripts.render_batch -debug -headless
Start Rendering thuman2 with 36 views, 512x512 size.
Output dir: ./debug/thuman2_36views
Rendering types: ['light', 'normal', 'depth']
  0%|                                                                                                                                                                    | 0/2 [00:04<?, ?it/s]Traceback (most recent call last):
  File "/home/ciciyang/miniconda3/envs/icon/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/ciciyang/miniconda3/envs/icon/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/ciciyang/code/ICON/scripts/render_batch.py", line 254, in <module>
    for _ in tqdm(
  File "/home/ciciyang/miniconda3/envs/icon/lib/python3.8/site-packages/tqdm/std.py", line 1180, in __iter__
    for obj in iterable:
  File "/home/ciciyang/miniconda3/envs/icon/lib/python3.8/multiprocessing/pool.py", line 868, in next
    raise value
multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7f8c12b3dfa0>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'_

**Comment by lywudi** [comment]
> > > > Good job! When I run the command of "python -m scripts.render_batch -debug -headless", it gives an error as following:干得好！当我运行“python -m scripts.render_batch -debug -headless”命令时，它给出了如下错误：
> > > > ```shell
> > > > Star /t Rendering thuman2 with 36 views, 512x512 size.
> > > > Output dir: ./debug/thuman2_36views
> > > > Rendering types: ['light', 'normal', 'depth']
> > > >   0%|                                                                                                                                                                                 | 0/2 [00:00<?, ?it/s]libEGL warning: DRI2: failed to authenticate
> > > > libEGL warning: DRI2: failed to authenticate
> > > > Compilation failure for vertex shader:
> > > > b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'
> > > > Compilation failure for fragment shader:
> > > > b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'
> > > > Linker failure: 
> > > > b'error: linking with uncompiled/unspecialized shadererror: linking with uncompiled/unspecialized shader'
> > > >   0%|                                                                                                                                                                                 | 0/2 [00:03<?, ?it/s]
> > > > Traceback (most recent call last):
> > > >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 194, in _run_module_as_main
> > > >     return _run_code(code, main_globals, None,
> > > >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/runpy.py", line 87, in _run_code
> > > >     exec(code, run_globals)
> > > >   File "/media/linjie/2473F3E9F7F20D6D/Code/ICON/scripts/render_batch.py", line 254, in <module>
> > > >     for _ in tqdm(
> > > >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/site-packages/tqdm/std.py", line 1195, in __iter__
> > > >     for obj in iterable:
> > > >   File "/home/linjie/anaconda3/envs/ICON/lib/python3.8/multiprocessing/pool.py", line 868, in next
> > > >     raise value
> > > > multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fe4a0bf7a30>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > How can I fix this?我该如何解决此问题？
> > > 
> > > 
> > > I've found the same issue here, directly use `export MESA_GL_VERSION_OVERRIDE=3.3` doesn't useful for me, but with os.environ is useful我在这里也发现了同样的问题，直接使用 `export MESA_GL_VERSION_OVERRIDE=3.3` 对我来说没有用，但与os.environ一起使用很有用
> > 
> > 
> > Hello, thanks for your kind sharing. Whether you use `os.environ` to solve `b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'` or another error `multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fe4a0bf7a30>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'` I am still looking for a solution to the second error. Could you please give me some help.你好，谢谢你的分享。无论你使用`os.environ`来解决 `b'0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.00 ES, and 3.00 ES\n'` 还是另一个错误 `multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fe4a0bf7a30>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'` ，我仍然在寻找第二个错误的解决方案。你能给我一些帮助吗？
> 
> Hey, I want to know if you have solved the second error. I also met the same error and have tried all the solutions above but it's not solved.我想知道你是否解决了第二个错误。我也遇到了同样的错误，并尝试了上述所有的解决方案，但它没有解决。 My error is as follows:我的错误如下： _(icon) ciciyang@DESKTOP-RVT20TR:~/code/ICON$ python -m scripts.render_batch -debug -headless_（icon）ciciyang@DESKTOP-RVT20TR：~/code/ICON$ python -m scripts.render_batch -debug -headless Start Rendering thuman2 with 36 views, 512x512 size.开始渲染thuman2，36个视图，512x512大小。 Output dir: ./debug/thuman2_36views输出目录：./调试/thuman2_36视图 Rendering types: ['light', 'normal', 'depth']渲染类型：“光线”、“正常”、“深度”] 0%| | 0/2 [00:04<?, ?it/s]Traceback (most recent call last):零百分比| | 0/2 [00：04<？，? it/s]Traceback（最后一次调用）： File "/home/ciciyang/miniconda3/envs/icon/lib/python3.8/runpy.py", line 194, in _run_module_as_main文件“/home/ciciyang/miniconda 3/envs/icon/lib/python3.8/runpy.py”，第194行，in _run_module_as_main return _run_code(code, main_globals, None,返回运行代码（代码，主全局，无， File "/home/ciciyang/miniconda3/envs/icon/lib/python3.8/runpy.py", line 87, in _run_code File“/home/ciciyang/miniconda3/envs/icon/lib/python3.8/runpy.py“，line 87，in run_code exec(code, run_globals) exec（代码，运行全局） File "/home/ciciyang/code/ICON/scripts/render_batch.py", line 254, in 文件“/home/ciciyang/code/ICON/scripts/render_batch.py”，第254行， for _ in tqdm( for _ in tqdm（ File "/home/ciciyang/miniconda3/envs/icon/lib/python3.8/site-packages/tqdm/std.py", line 1180, in **iter**File“/home/ciciyang/miniconda3/envs/icon/lib/python3.8/site-packages/tqdm/std.py“，line 1180，in**iter** for obj in iterable: 在iterable中： File "/home/ciciyang/miniconda3/envs/icon/lib/python3.8/multiprocessing/pool.py", line 868, in next文件“/home/ciciyang/miniconda 3/envs/icon/lib/python3.8/multiprocessing/pool.py“，第868行，在下一个 raise value 提高价值 multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7f8c12b3dfa0>'. Reason: 'ValueError('ctypes objects containing pointers cannot be pickled')'multiprocessing.pool.MaybeEncodingError：发送结果时出错："0x7f8c12b3dfa0>处的<multiprocessing.pool.ExceptionWithTraceback对象“。原因：'ValueError（'不能拾取包含指针的ctypes对象'）'_

您好。请问您解决这个问题了吗我最近在处理thuman数据集时，也遇到了同样的问题

**Comment by lywudi** [comment]
> > You could use the old rendering code, see the old [dataset.md](https://github.com/YuliangXiu/ICON/blob/74b5c7db64b04a13d5b921504921f2f4231119c5/docs/dataset.md) and old [scripits](https://github.com/YuliangXiu/ICON/tree/74b5c7db64b04a13d5b921504921f2f4231119c5/scripts)你可以使用旧的渲染代码，看到旧的 [数据集。md](https://github.com/YuliangXiu/ICON/blob/74b5c7db64b04a13d5b921504921f2f4231119c5/docs/dataset.md) 和老 [scripits](https://github.com/YuliangXiu/ICON/tree/74b5c7db64b04a13d5b921504921f2f4231119c5/scripts)
> 
> I'm sorry to bother you again.我很抱歉再打扰你。 I found the rendering process time-consuming. After browsing the script file, I find that two Gpus are used to run the script. I wonder if my understanding is correct.我发现呈现过程中耗费时间。 后浏览脚本文件时，我发现这两个集用于运行的脚本。 我不知道如果我的理解是正确的。

Do you solve this problem
